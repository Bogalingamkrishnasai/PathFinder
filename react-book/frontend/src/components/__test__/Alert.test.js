import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Alert from '../Alert';

// Mock CSS imports
jest.mock('../Alert.css', () => ({}));

describe('Alert Component', () => {
  test('renders children when visible', () => {
    render(<Alert visible type="success">Test message</Alert>);
    expect(screen.getByText('Test message')).toBeInTheDocument();
    expect(screen.getByRole('alert')).not.toHaveAttribute('hidden');
  });

  test('renders hidden when not visible', () => {  // Changed test name and expectation
    render(<Alert visible={false} type="success">Test message</Alert>);
    const alert = screen.getByRole('alert', { hidden: true });
    expect(alert).toHaveAttribute('hidden');
  });

  test('applies correct styles for success type', () => {
    render(<Alert visible type="success">Test</Alert>);
    const alert = screen.getByRole('alert');
    expect(alert).toHaveStyle('background-color: #adc6a8');
  });

  test('matches snapshot when visible', () => {
    const { asFragment } = render(<Alert visible type="success">Test message</Alert>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot when not visible', () => {  // Added new snapshot test
    const { asFragment } = render(<Alert visible={false} type="success">Test message</Alert>);
    expect(asFragment()).toMatchSnapshot();
  });
});