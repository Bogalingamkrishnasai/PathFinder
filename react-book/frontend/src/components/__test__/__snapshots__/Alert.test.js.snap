// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component matches snapshot when not visible 1`] = `
<DocumentFragment>
  <div
    class="alert-component false"
    hidden=""
    role="alert"
    style="background-color: rgb(173, 198, 168);"
  >
    Test message
  </div>
</DocumentFragment>
`;

exports[`Alert Component matches snapshot when visible 1`] = `
<DocumentFragment>
  <div
    class="alert-component visible"
    role="alert"
    style="background-color: rgb(173, 198, 168);"
  >
    Test message
  </div>
</DocumentFragment>
`;
